### ISSUE TITLE
Prometheus exporter for queue status

### ISSUE THREAD

**Issue by xrl** [rocketjob/semantic_logger#164]
I'm starting to "prometheus all the things" in our rails application. They have support for ruby garbage collector, rails, puma, activerecord, etc, would this project be open to support on the semantic logger operational metrics?

I'd love to get stats like 
 * Count of all log lines processed
 * Count of all rejected log lines
 * Size of queues
 * Thoughts on other properties to instrument?

I'm using: https://github.com/discourse/prometheus_exporter

I could author instrumentation for semantic logger but I'm not sure where to include it. I could put it in this code base? Or would you prefer it's a third party repo?

Perhaps this library could make a `SemanticLogger.stats` property like they do on Puma? I'm looking at https://github.com/puma/puma/blob/64c0153cd0bd2bb0fe85258e17dad538067c6484/lib/puma.rb#L28

**Comment by reidmorrison** [comment]
Yes, adding stats directly to Semantic Logger would be a useful addition. Just be sure that it is thread safe and does not slow down each of the concurrent appenders. Some appenders have their own queues and worker threads too.

**Comment by reidmorrison** [comment]
@xrl Operational stats have landed in #354. `SemanticLogger.stats` returns a Hash with the cumulative count of log entries processed, the count of dropped (rejected) entries, and queue sizes, both for the main pipeline and for each appender that runs its own async/batch thread. It is designed to be cheap to read and thread-safe, adding no locking to the logging hot path, so it should map straight onto a Prometheus exporter.

Docs are live here: https://logger.rocketjob.io/api.html#monitoring-the-background-log-thread

Would you mind taking a look and letting us know if the shape of the data works for your use case, or if you have any recommendations? Thanks for the suggestion.
